__tests__/utils/helpers.test.js

```diff
@@ -108,4 +133,94 @@ test('Buffer to hex', () => {
   const hexString = '044f355bdcb7cc0af728ef3cceb9615d90684bb5b2ca5f859ab0f0b704075871aa385b6b1b8ead809ca67454d9683fcf2ba03456d6fe2c4abe2b07f0fbdbb2f1c1';
   const buff = hexToBuffer(hexString);
   expect(bufferToHex(buff)).toBe(hexString);
-});
\ No newline at end of file
+});
+
+test('createTxFromData', () => {
+  const testnet = new Network('testnet');
+  // create token transaction
+  const createTokenTx = {
+      'name': 'test token',
+      'symbol': 'TST',
+      'tokens': ['01'],
+      'timestamp': dateFormatter.dateToTimestamp(new Date()),
+      'weight': 22.719884359974895,
+      'version': CREATE_TOKEN_TX_VERSION,
+      'inputs': [
+        {
+          'tx_id': '0000000110eb9ec96e255a09d6ae7d856bff53453773bae5500cee2905db670e',
+          'index': 0,
+          'data': Buffer.alloc(70),
+        }
+      ],
+      'outputs': [
+        {
+          'type': 'p2pkh',
+          'address': 'WZ7pDnkPnxbs14GHdUFivFzPbzitwNtvZo',
+          'value': 100,
+          'tokenData': 1,
+        }
+      ],
+  };
+  const createTx = helpers.createTxFromData(createTokenTx, testnet);
+  expect(createTx.getType()).toBe('Create Token Transaction');
+
+  // default tx
+  const defaultTxData = {
+      'tokens': [],
+      'timestamp': dateFormatter.dateToTimestamp(new Date()),
+      'weight': 22.719884359974895,
+      'version': DEFAULT_TX_VERSION,
+      'inputs': [
+        {
+          'tx_id': '0000000110eb9ec96e255a09d6ae7d856bff53453773bae5500cee2905db670e',
+          'index': 0,
+          'data': Buffer.alloc(70),
+        }
+      ],
+      'outputs': [
+        {
+          'address': 'WZ7pDnkPnxbs14GHdUFivFzPbzitwNtvZo',
+          'value': 100,
+          'tokenData': 0,
+        }
+      ],
+  };
+  const p2pkh = new P2PKH(new Address('WZ7pDnkPnxbs14GHdUFivFzPbzitwNtvZo'));
+  const defaultTx = helpers.createTxFromData(defaultTxData, testnet);
+  expect(defaultTx.getType()).toBe('Transaction');
+  expect(defaultTx.outputs[0].script.toString('hex')).toBe(p2pkh.createScript().toString('hex'))
+  // XXX: we should have a way to test that an output is P2PKH, P2SH or data
```

Comment:
We can use the method `parseScript` from the `output`, can't we? It used to support only p2pkh but you added the full support

Reply:
I think we can just remove this XXX now that you are testing the output type